sbnana/CAFAna/Systs/EnergySysts.h

```diff
@@ -0,0 +1,240 @@
+//ETW May 2020 Fresh version for SBN
+
+#pragma once
+
+#include "sbnana/CAFAna/Core/ISyst.h"
+#include "sbnana/CAFAna/Core/Utilities.h"
+#include "sbnana/CAFAna/Analysis/ExpInfo.h"
+#include "sbnanaobj/StandardRecord/Proxy/SRProxy.h"
+
+#include "TFile.h"
+#include "TH1.h"
+#include "TRandom3.h"
+
+#include <cassert>
+
+namespace ana
+{
+
+  enum class EnergyScaleSystTerm {
+    Constant,
+    Sqrt,
+    InverseSqrt
+  };
+
+  enum class EnergyScaleSystParticle {
+    Muon,
+    Hadron,
+    Neutron,
+    EM,
+    ChargedHadron
+  };
+
+  enum class EnergyScaleSystDetector {
+    SBND,
+    MicroBooNE,
+    ICARUS,
+    All
+  };
+
+  template<EnergyScaleSystTerm term, EnergyScaleSystParticle part, EnergyScaleSystDetector detector = EnergyScaleSystDetector::All>
+  class EnergyScaleSyst: public ISyst
+  {
+  public:
+    EnergyScaleSyst(double _uncertainty, std::string name) : 
+      ISyst(name, name), uncertainty(_uncertainty) {}
+    void Shift(double sigma, caf::SRSliceProxy *sr, double& weight) const override
+    {
+      double scale = uncertainty * sigma;
+      auto baseline_cut = [](double baseline){ 
+        auto all = (detector == EnergyScaleSystDetector::All);
+        auto nd = (detector == EnergyScaleSystDetector::SBND && baseline < 120);
```

Comment:
Do the new CAFs have a working `hdr.det` yet? If so, we should use it, if not we should make a `Var kDetector` somewhere so we don't have duplicates of this logic around.

Reply:
hdr.det seems to be filled in the 2021C files I have. But there is no value in SREnums for MicroBooNE (assuming that's the right place to look). I guess that can be kUNKNOWN for now?

Reply:
Probably worth adding a value there, though the chances of having microboone CAFs anytime soon are quite low.